tests/ui/single_component_path_imports.rs

```diff
@@ -0,0 +1,9 @@
+// run-rustfix
+// compile-flags: --edition 2018
+#![warn(clippy::single_component_path_imports)]
+
+use regex;
```

Comment:
Please also add a test for 

```suggestion
use regex;
use serde as edres;
```